src/Provider/ProductUrlProvider.php

```diff
@@ -124,9 +130,9 @@ private function getLocaleCodes(): array
         return $this->channelLocaleCodes;
     }
 
-    private function createProductUrl(ProductInterface $product): SitemapUrlInterface
+    private function createProductUrl(ProductInterface $product): UrlInterface
     {
-        $productUrl = $this->sitemapUrlFactory->createNew();
+        $productUrl = $this->sitemapUrlFactory->createNew(''); // todo bypassing this new constructor right now
```

Comment:
Hm.. can we find an alternative for this? Then we can get rid of the setter.

Reply:
As mentioned earlier I don't think we should remove the setters since they will provide extension points for later. With regards to this 'hack' I think it is okay for now, since it doesn't alter any functionality - it just circumvents our new, more stricter, approach. Therefore I have added a todo so we can fix it later then the provider will be refactored